kaitaistruct.py

```diff
@@ -125,6 +125,9 @@ def size(self):
     def read_s1(self):
         return KaitaiStream.packer_s1.unpack(self.read_bytes(1))[0]
 
+    def write_s1(self, data):
```

Comment:
Please use the same order as in the existing [Java runtime library](https://github.com/kaitai-io/kaitai_struct_java_runtime/blob/serialization/src/main/java/io/kaitai/struct/KaitaiStream.java#L316-L387) (this means read methods and write methods coupled together, not this scattered mix). Again, maintainability and consistency purposes.

Reply:
I don't get what you mean. Do you want me to move write functions away? So people rading the code couldn't compare them at a single glance and modify them without scrolling pages of other code just because some programmer in the single branch did so? Is there serious reason for this?

Reply:
> I don't get what you mean.

It doesn't matter, I'll make it clear to you. The methods in the Python runtime library **MUST** appear in this order:

```java
// Reading
/// Integer numbers
//// Signed
abstract public byte readS1();
///// Big-endian
abstract public short readS2be();
abstract public int readS4be();
abstract public long readS8be();

///// Little-endian
abstract public short readS2le();
abstract public int readS4le();
abstract public long readS8le();

//// Unsigned
abstract public int readU1();

///// Big-endian
abstract public int readU2be();
abstract public long readU4be();
public long readU8be() {...}

///// Little-endian
abstract public int readU2le();
abstract public long readU4le();
public long readU8le() {...}

/// Floating point numbers
//// Big-endian
abstract public float readF4be();
abstract public double readF8be();

//// Little-endian
abstract public float readF4le();
abstract public double readF8le();


// Writing
/// Integer numbers
//// Signed
abstract public void writeS1(byte v);
///// Big-endian
abstract public void writeS2be(short v);
abstract public void writeS4be(int v);
abstract public void writeS8be(long v);

///// Little-endian
abstract public void writeS2le(short v);
abstract public void writeS4le(int v);
abstract public void writeS8le(long v);

//// Unsigned
abstract public void writeU1(int v);
///// Big-endian
abstract public void writeU2be(int v);
abstract public void writeU4be(long v);
abstract public void writeU8be(long v);

///// Little-endian
abstract public void writeU2le(int v);
abstract public void writeU4le(long v);
abstract public void writeU8le(long v);

/// Floating point numbers
//// Big-endian
abstract public void writeF4be(float v);
abstract public void writeF8be(double v);

//// Little-endian
abstract public void writeF4le(float v);
abstract public void writeF8le(double v);
```

> Do you want me to move write functions away?

Yes, I do. 

> So people reading the code couldn't compare them at a single glance and modify them without scrolling pages of other code

Let's make two comparisons which two functions are more similar to each other. 

1.
```py
def read_s1(self):
    return KaitaiStream.packer_s1.unpack(self.read_bytes(1))[0]
def write_s1(self, data):
    return self.write_bytes(KaitaiStream.packer_s1.pack(data))
```

2.
```py
def read_s1(self):
        return KaitaiStream.packer_s1.unpack(self.read_bytes(1))[0]
def read_s2be(self):
        return KaitaiStream.packer_s2be.unpack(self.read_bytes(2))[0]
```

Why would _people_ want to compare completely different functions `read_s1()` and `write_s1()` for which the greatest similarity is in the name?

Reply:
Well if you say they must, they probably do. Because I feel like you are starting to harass me at this point. Why functions that operate on the same data type, use the same packer struct and do the opposite operations must be placed in a different parts of the code is beyond me.

Reply:
My point is actually quite simple: we can maintain the runtime libraries much easier when all of them use the same approach, logic and order of the functions, unless there's a very good reason not to do so (and different authors with different opinions and preferences is not among these reasons). One can simply split the screen with the Java runtime on the left and e.g. Python runtime on the right and see that there are exactly the same functions in both of runtimes.

If you believe that this established order doesn't make sense and you can come up with a much better one, feel free to open a [new issue](https://github.com/kaitai-io/kaitai_struct/issues/new) which will involve changing the order in all runtimes we have. There we can discuss it, but rather bring a bunch of solid arguments with you, because I'm still not convinced.

It's not possible to scatter chaos and confusion by using different approaches across runtimes for no reason. I can see that you're apparently not maintaining any such large project, so you just have to trust me, like it or not.